app/models/event_stream.rb

```diff
@@ -0,0 +1,38 @@
+class EventStream
+
+  PAGESIZE = 30
+  ROLLBACK = 2
+
+  def initialize(discussion_reader:, focused: nil)
+    @discussion_reader = discussion_reader
+    @discussion = discussion_reader.discussion
+    @focused = focused
+  end
+
+  def initial_events
+    Event.where(discussion: @discussion)
+         .where('sequence_id > ?', load_from_event)
+         .order(sequence_id: :asc)
+         .limit(PAGESIZE)
+  end
+
+  def focused_sequence_id
```

Comment:
Thing is, we want to preload the events _before_ the user clicks on a discussion. Because we already have the discussion locally so it loads instantly.. it would be best if we already had the events too.

So I imagine that we fetch discussions as per usual, then we grab relevant events for the top few discussions in the current view of the dashboard.